src/main/java/com/rcv/GuiConfigController.java

```diff
@@ -585,45 +585,45 @@ private boolean checkForSaveAndTabulate() {
 
   private void loadConfig(ContestConfig config) {
     clearConfig();
-
-    textFieldContestName.setText(config.getContestName());
-    textFieldOutputDirectory.setText(config.getOutputDirectory());
-    if (config.getContestDate() != null && !config.getContestDate().isEmpty()) {
-      datePickerContestDate.setValue(LocalDate.parse(config.getContestDate(), DATE_TIME_FORMATTER));
+    RawContestConfig rawConfig = config.getRawConfig();
+
+    OutputSettings outputSettings = rawConfig.outputSettings;
```

Comment:
Something is telling me that it's better to keep the original design here (using `ContestConfig`). Maybe this stems from me thinking that explicit is always better than implicit, even for unspecified "default" settings.

For example, say we have a default value of 2 for a setting in one version of the software. By using `ContestConfig`, it would actually populate this default value in the GUI, then force the user to re-save before running the tabulator and there's a clear record of what all the settings were. Then in a later update to the software we change the default value to 3. If users don't have that explicitly-defined (default) value in the config, I can imagine them being very confused as to why they can't replicate past results unless they've been following the patch notes closely. Does that make sense?

Of course, this leaves us with the problem of how to indicate "unlimited" for `maxSkippedRank`. I think maybe -1 could be a good value for that, and just note the significance in the documentation.

Reply:
I don't really like the idea of using -1 to signify unlimited. It's more intuitive to indicate no limit by just not specifying a limit. But I think we can handle this fine if we update the code so that we don't try to add the default value back when the user has cleared a non-required field.

Reply:
See what you think now.

Reply:
After reviewing this, it seems like your redesign works well to address this problem with one caveat: we have to be *very* careful not to change "if not supplied" behavior in future versions. Filed #211 to discuss potential safeguards against this.